### PR TITLE
Fix sharing text variety mime type

### PR DESCRIPTION
**Pull request by hoangdat**
(writing e2e test)

## Summary by CodeRabbit

* **New Features**
  * Expanded incoming share support for external plain text (including charset-suffixed MIME types), HTML-like text, vCard, calendar files, and mailto links, with correct composer prefill as body content or attachments.
* **Bug Fixes**
  * Prevented missed or duplicated shared content when shares arrive before the inbox/composer is ready.
  * Improved handling of empty/failed share payloads so later shares still work reliably.
  * Enhanced iOS URL/deep-link bridging for mailto sharing from cold and warm opens.
* **Tests**
  * Added integration and unit test coverage for share intake routing, timing, and buffering behavior.

Demo
iOS

https://github.com/user-attachments/assets/9664efd4-bb57-41c2-9089-51e93647c8d7

### CODE REVIEW

**Comment by Arsnael** [comment]
Docker image published for this PR: linagora/tmail-web-pr:4671

File: lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart
Comment (tddang-linagora) [inline lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart:1]: Should this break the app if the type is not text?

**Comment by Arsnael** [comment]
Docker image published for this PR: linagora/tmail-web-pr:4671

File: lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart
Comment (hoangdat) [reply lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart:1]: no this is the handling for `case SharedMediaType.text:`

**Review by tddang-linagora — DISMISSED** [review]

**Comment by hoangdat** [comment]
- [x] should handle `mailto:` also

**Comment by hoangdat** [comment]
@coderabbit review

**Comment by Arsnael** [comment]
Docker image published for this PR: linagora/tmail-web-pr:4671

File: lib/main/utils/email_receive_manager.dart [hunk 1]
```diff
@@ -8,15 +9,25 @@ class EmailReceiveManager {
...
+  /// still on the login screen — isn't dropped: the live stream itself
+  /// doesn't buffer events emitted before a listener attaches.
   void registerReceivingFileSharingStreamWhileAppClosed() {
     ReceiveSharingIntent.instance.getInitialMedia().then((value) {
       setPendingFileInfo(value);
 
       ReceiveSharingIntent.instance.reset();
     });
+
+    _mediaStreamSubscription ??= ReceiveSharingIntent.instance
```
Comment (dab246) [inline lib/main/utils/email_receive_manager.dart]: Live share can be cleared by empty initial-media result

What changed:

`registerReceivingFileSharingStreamWhileAppClosed()` now starts `getInitialMedia()` and registers the live `getMediaStream()` subscription in the same method. The unit test explicitly waits one microtask before emitting the live event:

```dart
await Future<void>.delayed(Duration.zero);
```

That wait avoids a real race in the implementation. If a live EventChannel event arrives before `getInitialMedia()` completes, the live event is added to `_pendingSharedFileInfo`; then `getInitialMedia()` can resolve to an empty list and call `setPendingFileInfo([])`, clearing the share that just arrived.

Risk:

This can drop the exact class of event the branch wants to protect: a user shares into a warm app while startup/login/home initialization is still settling. The composer will not open because the buffered share is replaced with an empty list before the mailbox subscribes.

Recommendation:

Do not let an empty initial-media response overwrite a non-empty live share. Also add a deterministic test where `getInitialMedia()` completes after the live event.

Example direction:

```dart
void registerReceivingFileSharingStreamWhileAppClosed() {
  _ensurePendingSubjectOpen();

  _mediaStreamSubscription ??= ReceiveSharingIntent.instance
      .getMediaStream()
      .listen(
        setPendingFileInfo,
        onError: (error, stackTrace) {
          logWarning(
            'EmailReceiveManager::registerReceivingFileSharingStreamWhileAppClosed: '
            'getMediaStream error = $error',
          );
        },
      );

  ReceiveSharingIntent.instance.getInitialMedia().then((value) {
    if (value.isNotEmpty) {
      setPendingFileInfo(value);
    }
    return ReceiveSharingIntent.instance.reset();
  }).catchError((error, stackTrace) {
    logWarning(
      'EmailReceiveManager::registerReceivingFileSharingStreamWhileAppClosed: '
      'getInitialMedia error = $error',
    );
  });
}
```

File: lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart [hunk 2]
```diff
@@ -660,15 +659,6 @@ class MailboxDashBoardController extends ReloadableController
...
   void _registerReceivingFileSharingStream() {
-    _receivingFileSharingStreamSubscription = _emailReceiveManager
-      .receivingFileSharingStream
-      .listen(
-        _emailReceiveManager.setPendingFileInfo,
-        onError: (err) {
-          logWarning('MailboxDashBoardController::_registerReceivingFileSharingStream::receivingFileSharingStream:Exception = $err');
-        },
-      );
-
```
Comment (dab246) [inline lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart:671]: Closing the manager subject from dashboard can break later share delivery

What changed:

The live media subscription moved from `MailboxDashBoardController` to singleton `EmailReceiveManager`, but dashboard still closes the manager's `BehaviorSubject` in `onClose()`:

```dart
_emailReceiveManager.closeEmailReceiveManagerStream();
```

Meanwhile `_mediaStreamSubscription` is not cancelled. After a dashboard close, a new dashboard can subscribe to the currently closed `pendingSharedFileInfo`. When a later live share arrives, `_clearPendingFileInfo()` replaces `_pendingSharedFileInfo` with a new `BehaviorSubject`, but the dashboard listener remains attached to the old closed subject.

Risk:

After logout/login, route recreation, or any dashboard lifecycle close/open, external shares can stop opening composer even though the manager is still listening to the platform stream. This is a user-visible regression in the share intent flow.

Recommendation:

Keep `pendingSharedFileInfo` stable for the lifetime of `EmailReceiveManager`; do not close it from `MailboxDashBoardController`. If the goal is only to clear consumed data, expose a clear method that emits `[]` without closing the subject.

Example direction:

```dart
void clearPendingFileInfo() {
  if (!_pendingSharedFileInfo.isClosed) {
    _pendingSharedFileInfo.add(<SharedMediaFile>[]);
  }
}

Future<void> dispose() async {
  await _mediaStreamSubscription?.cancel();
  _mediaStreamSubscription = null;
  await _pendingSharedFileInfo.close();
}
```

Then dashboard should cancel only its own subscription:

```dart
if (PlatformInfo.isMobile) {
  _pendingSharedFileInfoSubscription?.cancel();
  _emailReceiveManager.clearPendingFileInfo();
  _deepLinkDataStreamSubscription?.cancel();
}
```

File: lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart [hunk 3]
```diff
@@ -704,7 +694,9 @@ class MailboxDashBoardController extends ReloadableController
             openComposer(
               ComposerArguments.fromFileShared([sharedMediaFile]),
             );
-          } else if (sharedMediaFile.mimeType == Constant.textPlainMimeType) {
```
Comment (dab246) [inline lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart:707]: Parameterized vCard mime types are treated as body text

What changed:

For `SharedMediaType.text`, the branch changed from only accepting exact `text/plain` to treating every non-vCard text share as body content:

```dart
if (sharedMediaFile.mimeType == Constant.textVCardMimeType) {
  openComposer(ComposerArguments.fromFileShared([sharedMediaFile]));
} else {
  openComposer(ComposerArguments.fromContentShared(sharedMediaFile.path.trim()));
}
```

Risk:

`Constant.textVCardMimeType` is `text/x-vcard`. Real MIME values can include parameters or case differences, for example `text/x-vcard;charset=utf-8`. Those values now fall into the `else` branch and become email body content instead of a shared contact file.

Recommendation:

Normalize MIME type before branching:

```dart
final normalizedMimeType = sharedMediaFile.mimeType
    ?.split(';')
    .first
    .trim()
    .toLowerCase();

if (normalizedMimeType == Constant.textVCardMimeType) {
  openComposer(ComposerArguments.fromFileShared([sharedMediaFile]));
} else {
  openComposer(ComposerArguments.fromContentShared(sharedMediaFile.path.trim()));
}
```

Suggested tests:

- `text/x-vcard;charset=utf-8` opens `composeFromFileShared`.
- `Text/Plain; Charset=UTF-8` opens `composeFromContentShared`.
- `text/html;charset=utf-8` opens `composeFromContentShared`.

File: test/main/utils/email_receive_manager_test.dart [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+  setUp(() {
+    messenger.setMockMethodCallHandler(messagesChannel, (call) async => null);
+  });
+
+  tearDown(() {
+    messenger.setMockMethodCallHandler(messagesChannel, null);
+  });
+
+  group('EmailReceiveManager', () {
```
Comment (dab246) [inline test/main/utils/email_receive_manager_test.dart:36]: New coverage misses lifecycle and race edge cases

What is covered:

- Live EventChannel share is buffered when manager is already registered and initial-media round trip has settled.
- `BehaviorSubject` replays a previously buffered event to a late subscriber.
- Android integration scenarios cover mailto before mailbox ready and text shares with charset/html mime types.

What is missing:

- Live event arrives before `getInitialMedia()` completes.
- Dashboard closes and is recreated before another share arrives.
- EventChannel emits an error.
- vCard text share includes MIME parameters.

Risk:

The current tests pass while the two critical lifecycle bugs above remain possible. The coverage is directionally good, but it avoids the sharp timing edges.

Recommendation:

Add focused manager/controller unit tests for these edge cases before merging. They are cheaper and more deterministic than Patrol tests.

Comment (hoangdat) [reply test/main/utils/email_receive_manager_test.dart:36]: added

File: lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart (hunk 3 above)
Comment (hoangdat) [reply lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart:707]: fixed

File: lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart (hunk 2 above)
Comment (hoangdat) [reply lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart:671]: added

**Comment by Arsnael** [comment]
Docker image published for this PR: linagora/tmail-web-pr:4671

**Review by tddang-linagora — DISMISSED** [review]

**Comment by Arsnael** [comment]
Docker image published for this PR: linagora/tmail-web-pr:4671

**Comment by hoangdat** [comment]
https://github.com/user-attachments/assets/591bbd0c-0315-4e57-a4fe-7102921edbd1


https://github.com/user-attachments/assets/6dee5e45-368b-4d63-bf72-aad11d145a15

**Comment by hoangdat** [comment]
### Mailto

https://github.com/user-attachments/assets/a0eb29c9-1638-4aa1-aa51-2bde0351577b


### Share link 

https://github.com/user-attachments/assets/fc0d1e47-3b00-4d21-b25a-19d88c5106ce

**Review by dab246 — APPROVED** [review]
LGTM
